### PR TITLE
docs: add coder login to CI docs

### PR DESCRIPTION
**Pull request by ericpaulsen**
a customer asked how to non-interactively authenticate the `coder templates push` command we have in our CI documentation. this PR documents this step by adding the `coder login <url> --token $CODER_SESSION_TOKEN` command.

### CODE REVIEW

**Review by matifali — DISMISSED** [review]

**Comment by matifali** [comment]
@ericpaulsen we don't need to login because we already have the `$CODER_SESSION_TOKEN`
`coder templates push` work without a need to login. So I don't think we need to add this to docs.

File: docs/templates/change-management.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ export CODER_TEMPLATE_DIR=.coder/templates/kubernetes
 export CODER_TEMPLATE_VERSION=$(git rev-parse --short HEAD)
 
 # Push the new template version to Coder
+coder login <access-url> --token $CODER_SESSION_TOKEN
```
[the remark this answers is not in the document]
Comment (ericpaulsen) [reply docs/templates/change-management.md]: @matifali this is not working for me:

```console
➜  coder templates push auth-test
running command "coder templates push": You are not logged in. Try logging in using 'coder login <url>'.:
    github.com/coder/coder/cli.init                                                                      
        /home/runner/actions-runner/_work/coder/coder/cli/root.go:74                                     

✗  echo $CODER_SESSION_TOKEN
KbqY3sl...hGS8lPj
 
➜  echo $CODER_URL
eric.oss.demo.coder.com
```

**Comment by ericpaulsen** [comment]
@matifali this worked for me:

```
coder templates create auth-test --token $CODER_SESSION_TOKEN --url $CODER_URL
```

although our docs do not include the `--token` and `--url` flags in the `coder templates push` command. does this need to be corrected?

**Comment by matifali** [comment]
@ericpaulsen they are mot required. Coder cli can automatically read these environment variables. 
also if you are passing them as —url and —token then you don’t need have those environment variables. 

1st approach 

```shell
coder templates create auth-test --token some-token --url dev.coder.com
```

2nd approach
```shell
coder templates create auth-test 
```
given `$CODER_SESSION_TOKEN` and `$CODER_URL` are available

**Review by sharkymark — APPROVED** [review]
you could also pass the --url for each coder command along with --token

**Comment by ericpaulsen** [comment]
@matifali the 1st approach does not work for me, as shown in this [comment](https://github.com/coder/coder/pull/9038#discussion_r1291554898). i will push a commit with the `--url` and `--token` flags.

File: docs/templates/change-management.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (matifali) [reply docs/templates/change-management.md]: How are you setting the variables. You need to use
```shell
export $CODER_SESSION_TOKEN=KbqY3sl...hGS8lPj
export $CODER_URL=eric.oss.demo.coder.com
coder templates create auth-test
```

**Pull request merged**